internal/pkg/argocd/argocd.go

```diff
@@ -410,6 +410,36 @@ func createTempAppObjectFroNewApp(ctx context.Context, componentPath string, rep
 	}
 }
 
+// deleteTempAppObject deletes a temporary app object created for diff generation
+// returns an error if the deletion fails
+func deleteTempAppObject(ctx context.Context, ac argoCdClients, app *argoappv1.Application) error {
```

Comment:
not happy with the temp in the function name but it's following the `createTempAppObjectFroNewApp` naming (which has a typo 😄 )